salt/modules/zypper.py

```diff
@@ -1768,17 +1823,52 @@ def search(criteria, refresh=False):
     if refresh:
         refresh_db()
 
-    solvables = __zypper__.nolock.xml.call('se', criteria).getElementsByTagName('solvable')
+    cmd = ['se']
+    if kwargs.get('match') == 'exact':
+        cmd.append('--match-exact')
+    elif kwargs.get('match') == 'words':
+        cmd.append('--match-words')
+    elif kwargs.get('match') == 'substrings':
+        cmd.append('--match-substrings')
+
+    if kwargs.get('provides'):
+        cmd.append('--provides')
+    if kwargs.get('recommends'):
+        cmd.append('--recommends')
+    if kwargs.get('requires'):
+        cmd.append('--requires')
+    if kwargs.get('suggests'):
+        cmd.append('--suggests')
+    if kwargs.get('conflicts'):
+        cmd.append('--conflicts')
+    if kwargs.get('obsoletes'):
+        cmd.append('--obsoletes')
+
+    if kwargs.get('file_list'):
+        cmd.append('--file-list')
+    if kwargs.get('search_descriptions'):
+        cmd.append('--search-descriptions')
+    if kwargs.get('case_sensitive'):
+        cmd.append('--case-sensitive')
+    if kwargs.get('installed_only'):
+        cmd.append('--installed-only')
+    if kwargs.get('not_installed_only'):
+        cmd.append('--not-installed-only')
+    if kwargs.get('details'):
+        cmd.append('--details')
```

Comment:
I would not do all that code. Simply because more to remember and few dozen of codelines. Why?.. I would actually use "-" as "_" and reuse original Zypper parameters. Then you would just do only this (and nothing else):
```python
for opt in kwargs.keys():
    cmd.append('--{0}'.format(opt.replace('_', '-'))
```
And you would be able to just read manual of Zypper and do not need to memorize all that mapping between two tools. That's it.

Reply:
I do not like to be implicit. And it would remove the possibility to just not allow some combination
of options like the match one.
It would also require that the user must understand the algorithm or we at least document the
possibilities.

Reply:
>  it would remove the possibility to just not allow some combination

```python
for opt in kwargs:
    opt = opt.replace('_', '-')
    if opt in ALLOWED_ARRAY_OF_COMMANDS:
        cmd.append('--{0}'.format(opt))
```

> It would also require that the user must understand the algorithm

What algorithm? You're right now just proxy-adding options from Salt's `pkg.<function> <options>` to `zypper <function> <options>` and nothing else. So I would prefer minimum code here.